frontend/app/src/components/wrapped/WrappedContainer.vue

```diff
@@ -45,13 +46,35 @@ const end = ref('');
 const start = ref('');
 const summary = ref<WrapStatisticsResult>();
 
+const { fetchHistoryEvents } = useHistoryEvents();
+
 const { isFirstLoad, loading: sectionLoading } = useStatusUpdater(Section.HISTORY_EVENT);
 const eventTaskLoading = isTaskRunning(TaskType.TRANSACTIONS_DECODING);
 const protocolCacheUpdatesLoading = isTaskRunning(TaskType.REFRESH_GENERAL_CACHE);
 const onlineHistoryEventsLoading = isTaskRunning(TaskType.QUERY_ONLINE_EVENTS);
 
 const refreshing = logicOr(sectionLoading, eventTaskLoading, onlineHistoryEventsLoading, protocolCacheUpdatesLoading);
 
+const historyEventsReady = logicAnd(!isFirstLoad(), logicNot(refreshing));
+const debouncedHistoryEventsReady = debouncedRef(historyEventsReady, 500);
+const usedHistoryEventsReady = logicAnd(historyEventsReady, debouncedHistoryEventsReady);
+
+watchImmediate(usedHistoryEventsReady, async (curr, old) => {
+  if (curr && !old && get(start) === '') {
+    const response = await fetchHistoryEvents({
+      ascending: [true],
+      groupByEventIds: true,
+      limit: 1,
+      offset: 0,
+      orderByAttributes: ['timestamp'],
+    });
+
+    if (response.data.length === 1) {
+      set(start, convertFromTimestamp(Math.floor(response.data[0].timestamp / 1000)));
+    }
```

Comment:
do you think we can put most of the logic to `useHistoryEvents` as `getEarliestEventTimestamp(): number | undefined`?